infinite_tracing/test/support/fake_trace_observer.rb

```diff
@@ -87,6 +87,20 @@ def record_span(record_spans)
       end
     end
 
+    class FailedPreconditionInfiniteTracer < BaseInfiniteTracer
+      def initialize
+        super
+        @count = 0
+      end
+
+      def record_span(record_spans)
+        @lock.synchronize { @noticed.signal }
+        msg = "I don't exist!"
+        @count += 1
+        raise GRPC::BadStatus.new(GRPC::Core::StatusCodes::FAILED_PRECONDITION, msg) if @count <= 5
```

Comment:
This was the best way i could figure out how to limit how many times this goes through currently. Does this approach seem reasonable for now?

Reply:
This is perfect!